lib/Core/Executor.cpp

```diff
@@ -2566,16 +2576,92 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
     break;
   }
 #endif
+  case Instruction::InsertElement: {
+    InsertElementInst *iei = cast<InsertElementInst>(i);
+    ref<Expr> vec = eval(ki, 0, state).value;
+    ref<Expr> newElt = eval(ki, 1, state).value;
```

Comment:
Should we add a check that the type of `newElt` is consistent with the type of the vector elements?